### DISCUSSION TITLE
Hetzner sync ?

### DISCUSSION THREAD

**Discussion by SerhiiZahuba** [bb-Ricardo/netbox-sync#286] [Ideas]
Hello
Like your product very much. Do you have any plans to sync with Hetzner?

https://docs.hetzner.cloud/#servers

I will be very grateful to you.

**Comment by bb-Ricardo** [comment]
Hi,

This is currently not implemented but could be added with another source.

Maybe we could ask the Hetzner guys if they are willing to provide an implementation?

**Comment by SerhiiZahuba** [comment]
They don't really want to talk about it

If you have some time and desire - I will be very grateful.
Below is the documentation on the subject
1) Their python library
https://github.com/hetznercloud/hcloud-python

Authorization example
curl \
	-H "Authorization: Bearer $API_TOKEN" \
	'https://api.hetzner.cloud/v1/actions'
    
Example Response:
```json    
{
    "servers": [],
    "meta": {
        "pagination": {
            "page": 1,
            "per_page": 25,
            "previous_page": null,
            "next_page": null,
            "last_page": 1,
            "total_entries": 0
        }
    }
}
```

Response sample:
```json
{
  "actions": [
    {
      "command": "start_server",
      "error": {
        "code": "action_failed",
        "message": "Action failed"
      },
      "finished": "2016-01-30T23:55:00+00:00",
      "id": 42,
      "progress": 100,
      "resources": [
        {
          "id": 42,
          "type": "server"
        }
      ],
      "started": "2016-01-30T23:55:00+00:00",
      "status": "success"
    }
  ],
  "meta": {
    "pagination": {
      "last_page": 4,
      "next_page": 4,
      "page": 3,
      "per_page": 25,
      "previous_page": 2,
      "total_entries": 100
    }
  }
}
```

**Reply by bb-Ricardo** [reply]
Hi, I'm sorry but currently not possible to be implemented by myself. Just taking care of the current issues and my other projects takes already quite some time.
